components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/dao/BPSProfileDAO.java

```diff
@@ -151,11 +209,24 @@ public BPSProfileDTO getBPSProfile(String profileName, int tenantId, boolean isW
                 String password = rs.getString(SQLConstants.PASSWORD_COLUMN);
                 String callbackUser = rs.getString(SQLConstants.CALLBACK_USER_COLUMN);
                 String callbackPassword = rs.getString(SQLConstants.CALLBACK_PASSWORD_COLUMN);
+
+                try {
+                    CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
+                    byte[] decryptedPasswordBytes = cryptoUtil.base64DecodeAndDecrypt(password);
+                    decryptPassword = new String(decryptedPasswordBytes, "UTF-8");
+                    byte[] decryptedCallBackPasswordBytes = cryptoUtil.base64DecodeAndDecrypt(callbackPassword);
+                    decryptCallBackPassword = new String(decryptedCallBackPasswordBytes, "UTF-8");
+
+                } catch (CryptoException | UnsupportedEncodingException e) {
+                    throw new InternalWorkflowException("Error while decrypting the password for BPEL Profile" + " " +
+                            profileName, e);
+                }
+
```

Comment:
Better to move this code to a private method or util method since this code is duplicated at several locations